service/worker/pernamespaceworker.go

```diff
@@ -194,13 +194,17 @@ func (wm *perNamespaceWorkerManager) namespaceCallback(ns *namespace.Namespace,
 	go wm.getWorkerByNamespace(ns).refreshWithNewNamespace(ns, deleted)
 }
 
+func (wm *perNamespaceWorkerManager) refreshAll() {
+	wm.lock.Lock()
+	defer wm.lock.Unlock()
+	for _, worker := range wm.workers {
+		go worker.refreshWithExistingNamespace()
```

Comment:
is it possible that refershAll get called multiple times before refreshWithExistingNamespace() finished and end up having multiple goroutines running refreshWithExistingNamespace()?

Reply:
Yes, that's possible before this change too. The refresh* functions are written so that it's ok, they use a lock and some state to synchronize. The worst that should happen is extra calls to get dynamicconfig values